src/DynamoFeatureFlags/FeatureFlagsManager.cs

```diff
@@ -0,0 +1,144 @@
+            if(mobileKey == null)
+            {
+                //load key from config depending on build config.
+#if DEBUG
+                var keystring = "ldmobilekey_dev";
+#else
+                var keystring = "ldmobilekey_prd";
+#endif
+                //don't use configuration manager as it is not net std 2 compliant.
+                var path = $"{ GetType().Assembly.Location}.config";
+                var key = GetConfigurationItem(keystring, path);
+               
+                if (key != null)
+                {
+                    mobileKey = key;
+                }
+            }
+            //if mobile key is still null after loading from config, bail.
+            if(mobileKey == null)
+            {
+                throw new ArgumentException("ld mobile key was null");
+            }
+            if (userkey == null)
+            {
+                MessageLogged?.Invoke("The userkey was null when starting feature flag manager, using a shared key," +
+                    " possibly analytics was disabled, test mode is active, or headless mode is active.");
+                userkey = sharedUserKey;
+            }
+            //send user as anonymous//https://docs.launchdarkly.com/home/users/anonymous-users
+            user = LaunchDarkly.Sdk.User.Builder(userkey).Anonymous(true).Build();
+
+            Init(mobileKey);
+        }
+        internal async void Init(string mobileKey)
+        {
+            //start up client.
+            ldClient = await LaunchDarkly.Sdk.Client.LdClient.InitAsync(mobileKey, user);
+            if (ldClient.Initialized)
+            {
+                MessageLogged?.Invoke($"launch darkly initalized");
+            }
+            else
+            {
+                MessageLogged?.Invoke($"launch darkly failed to initalize");
+            }
+        }
+
+        // TODO as we need more flag types, implement cases here or break out
+        // into more specific methods.
+        /// <summary>
+        /// Check the value of a specific flag. If the internal client is not initialized yet, will simply return the default value you provide.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="flagkey">string key of the flag to lookup.</param>
+        /// <param name="defaultval">value to return if there is an error reading the flag value.</param>
+        /// <returns></returns>
+        internal static T CheckFeatureFlag<T>(string flagkey,T defaultval)
+        {
+            try
+            {
+                if (ldClient == null || !ldClient.Initialized)
+                {
+                    MessageLogged($"feature flags client not initalized for requested flagkey: {flagkey}, returning default value: {defaultval}");
+                    return defaultval;
+                }
+
+                object output = default(T);
+                switch (default(T))
+                {
+                    case bool _:
```

Comment:
Are we testing for type here? If so, can we also use `typeof` as in:
```
switch (typeof(T))
{
    case typeof(bool):
...
}
```

Reply:
![Screen Shot 2022-04-18 at 11 58 27 AM](https://user-images.githubusercontent.com/508936/163836063-c38ff2b3-9c28-4eb1-893a-7d82e4c786db.png)

also fails on dotnet 6 unfortunately, mixing generic and switch case is not a great idea I guess.